client/client-http/src/main/java/software/amazon/smithy/java/client/http/HttpErrorDeserializer.java

```diff
@@ -362,5 +409,18 @@ public Builder unknownErrorFactory(UnknownErrorFactory unknownErrorFactory) {
             this.unknownErrorFactory = Objects.requireNonNull(unknownErrorFactory, "unknownErrorFactory is null");
             return this;
         }
+
+        /**
+         * The parser to parse the shapeId from the payload.
+         *
+         * <p>By default, payload is parsed into a JSON document
```

Comment:
Hm that's not right. It's parsed into a document I think, and documents are not == JSON. Please also expand on how it identifies an error by using the discriminator of a document.